File: src/test/java/uk/co/eelpieconsulting/feedlistener/controllers/FeedItemPopulatorTest.java
package uk.co.eelpieconsulting.feedlistener.controllers;

import org.joda.time.DateTime;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;
import uk.co.eelpieconsulting.feedlistener.controllers.ui.SubscriptionLabelService;
import uk.co.eelpieconsulting.feedlistener.model.FeedItem;

import java.util.UUID;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class FeedItemPopulatorTest {

    private SubscriptionLabelService subscriptionLabelService = Mockito.mock(SubscriptionLabelService.class);

    @Test
    public void canCorrectForExcessivelyEscapedInputFeeds() {
        final String incorrectlyEscapedHeadline = "St John&#39;s Bar, 5 Cable Street, Te Aro, Wellington";
        FeedItem feedItem = new FeedItem(incorrectlyEscapedHeadline, "http://localhost", null, DateTime.now().toDate(), null, null, null,
                UUID.randomUUID().toString(), UUID.randomUUID().toString());

        FeedItem fixed = new FeedItemPopulator(subscriptionLabelService).overlyUnescape(feedItem);

        assertEquals("St John's Bar, 5 Cable Street, Te Aro, Wellington", fixed.getHeadline());
    }

}

File: src/test/kotlin/uk/co/eelpieconsulting/feedlistener/controllers/FeedItemPopulatorTest.kt
package uk.co.eelpieconsulting.feedlistener.controllers

import org.bson.types.ObjectId
import org.joda.time.DateTime
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.mockito.Mockito
import uk.co.eelpieconsulting.feedlistener.controllers.ui.SubscriptionLabelService
import uk.co.eelpieconsulting.feedlistener.model.FeedItem
import java.util.*

class FeedItemPopulatorTest {
    private val subscriptionLabelService = Mockito.mock(
        SubscriptionLabelService::class.java
    )

    @Test
    fun canCorrectForExcessivelyEscapedInputFeedItems() {
        val incorrectlyEscapedHeadline = "St John&#39;s Bar, 5 Cable Street, Te Aro, Wellington"
        val feedItem = FeedItem(
            objectId = ObjectId.get(),
            title = incorrectlyEscapedHeadline, url = "http://localhost", date = DateTime.now().toDate(), body = null,
            subscriptionId = UUID.randomUUID().toString(), channelId = UUID.randomUUID().toString(), _categories = emptyList(), ordering = DateTime.now().toDate()
        )
        val fixed = FeedItemPopulator(subscriptionLabelService).overlyUnescape(feedItem)
        Assertions.assertEquals("St John's Bar, 5 Cable Street, Te Aro, Wellington", fixed.headline)
    }

}
